server/src/main/java/fr/uga/l3miage/tp4/models/UserEntity.java

```diff
@@ -0,0 +1,19 @@
+package fr.uga.l3miage.tp4.models;
+
+import lombok.Getter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+@Getter
+@Entity
+public abstract class UserEntity {
+    @Id
+    private Long id;
+    private String firstname;
+    private String lastname;
+    @Column(unique = true, nullable = false)
+    private String email;
+    @Column(unique = true)
+    private String phoneNumber;
+}
```

Comment:
ici vous n'avez pas d'héritage
```suggestion
@MappedSuperclass
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
public abstract class UserEntity {
    @Id
    private Long id;
    private String firstname;
    private String lastname;
    @Column(unique = true, nullable = false)
    private String email;
    @Column(unique = true)
    private String phoneNumber;
}
```